app/controllers/api/v2/user_columns_controller.rb

```diff
@@ -0,0 +1,21 @@
+module Api
+  module V2
+    class UserColumnsController < V2::BaseController
+      include Api::Version2
+
+      api :GET, "/user_columns/", N_("List of User columns for a user")
+      param :resource, String, :required => true
+      def index
+        User.current.user_columns.where(:resource => params[:resource])
+      end
+
+      api :POST, "/user_columns/", N_("Creates a user column for a given resource")
+      param :resource, String, :required => true
+      param :columns, Array, :desc => N_("List of user selected columns")
+      def create
+        User.current.user_columns.where(:resource => params[:resource]).destroy_all
```

Comment:
CREATE "updates" via this destroy_all?

Reply:
Good point will figure out a better way.